helpers/chatHelper.py

```diff
@@ -266,462 +264,623 @@ async def partChannel(
+        but=[user_token["token_id"]],
+    )
+
+
+async def handle_private_message(
+    user_token: osuToken.Token,
+    send_to: str,
+    message: str,
+) -> Optional[int]:
+    recipient_token = await tokenList.getTokenFromUsername(send_to)
+    if recipient_token is None:
         logger.warning(
-            "Sender or recipient in messaging interaction were restricted",
+            "User attempted to send a message to an unknown user",
             extra={
-                "username": userToken["username"],
-                "user_id": userToken["user_id"],
-                "recipient": to,
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient": send_to,
             },
         )
-        return 404
-    except exceptions.userTournamentException:
+        return ...
+
+    if recipient_token["tournament"]:
         logger.warning(
-            "User tried to send a message to a tournament client",
+            "User attempted to send a message to a tournament client",
             extra={
-                "username": userToken["username"],
-                "user_id": userToken["user_id"],
-                "recipient": to,
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient": send_to,
             },
         )
-        return 404
-    except exceptions.userNotFoundException:
-        logger.warning("User not connected to IRC/Bancho.")
-        return 401
-    except exceptions.userBlockingDMsException:
+        return ...
+
+    if await osuToken.isSilenced(recipient_token["token_id"]):
+        await osuToken.enqueue(
+            user_token["token_id"],
+            serverPackets.targetSilenced(recipient_token["username"]),
+        )
+
+        logger.warning(
+            "User tried to send a message to a silenced user",
+            extra={
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+
+        return ...
+
+    if osuToken.is_restricted(recipient_token["privileges"]):
+        logger.warning(
+            "User attempted to send a message to a restricted user",
+            extra={
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+        return ...
+
+    # TODO: check if user has us blacklisted
+    if (  # type: ignore
```

Comment:
a bit curious as to why type ignore is needed here

Reply:
Resolved in 351f389